egui/src/widgets/plot/mod.rs

```diff
@@ -20,12 +20,43 @@ mod items;
 mod legend;
 mod transform;
 
-type CustomLabelFunc = dyn Fn(&str, &Value) -> String;
-type CustomLabelFuncRef = Option<Box<CustomLabelFunc>>;
-
-type AxisFormatterFn = dyn Fn(f64) -> String;
+type LabelFormatterFn = dyn Fn(&str, &Value) -> String;
+type LabelFormatter = Option<Box<LabelFormatterFn>>;
+type AxisFormatterFn = dyn Fn(f64, &RangeInclusive<f64>) -> String;
 type AxisFormatter = Option<Box<AxisFormatterFn>>;
 
+/// Specifies the coordinates formatting when passed to [`Plot::coordinates_formatter`].
+pub struct CoordinatesFormatter {
+    function: Box<dyn Fn(&Value) -> String>,
+}
+
+impl CoordinatesFormatter {
+    pub fn new(function: impl Fn(&Value) -> String + 'static) -> Self {
+        Self {
+            function: Box::new(function),
+        }
+    }
+
+    /// Show a fixed number of decimal places.
+    pub fn with_precision(precision: usize) -> Self {
+        Self {
+            function: Box::new(move |value| {
+                format!("x: {:.p$}\ny: {:.p$}", value.x, value.y, p = precision)
+            }),
+        }
+    }
```

Comment:
Here (and in other places) it would be great to have a function to format with a fixed number of significant digits. Does egui happen to already have this functionality somewhere?